src/libponyrt/actor/actor.c

```diff
@@ -418,9 +430,12 @@ bool ponyint_actor_run(pony_ctx_t* ctx, pony_actor_t* actor, bool polling)
       // in the future we will wait for the CD to reach out and ask
       // if we're blocked or not.
       // But, only if gc.rc > 0 because if gc.rc == 0 we are a zombie.
-      set_flag(actor, FLAG_BLOCKED_SENT);
-      set_flag(actor, FLAG_CD_CONTACTED);
-      ponyint_cycle_block(actor, &actor->gc);
+      send_block(ctx, actor);
+
+      // Try and run GC because we're blocked and sending a block message
+      // to the CD and we're past the normal point at which `try_gc` is
+      // called as part of this function
```

Comment:
Can you change this message to more of "the why" rather than the "what".

Something like "we will try to free any memory we can as we are blocked and might hold on to that memory for a long time."

Reply:
updated